cocotb/triggers.py

```diff
@@ -696,7 +696,7 @@ class Combine(_AggregateWaitable):
 
     async def _wait(self):
         waiters = []
-        e = Event()
+        e = Event(self)
```

Comment:
I don't think the abuse is a big issue, this object cannot directly leak to the user, but I think this transition could be smoother so we don't see the `Event` at all, it is an implementation detail. I think that would require a special subclass of `Event`. Sounds like a follow up PR.